frontend/app/page.tsx

```diff
@@ -1,103 +1,176 @@
-import Image from "next/image";
+'use client'
+import React, { useState, FormEvent, useRef, useEffect } from "react";
+import styles from './style/home.module.css';
+
+interface Message {
+  role: 'user' | 'assistant';
+  content: string;
+  isError?: boolean; 
+}
+
 
 export default function Home() {
+
+  const [messages, setMessages] = useState<Message[]>([])
+  const [input, setInput] = useState('')
+  const [loading, setLoading] = useState(false)
+
+  const chatContainerRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (chatContainerRef.current) {
+      chatContainerRef.current.scrollTop = chatContainerRef.current.scrollHeight
+    }
+  }, [messages])
+
+  const handleSubmit = async (e: FormEvent) => {
+    e.preventDefault();
+    if (!input.trim() || loading) return;
+
+    const userQuery = input;
+    setInput('');
+    setLoading(true)
+
+    const newUserMessage: Message = { role: 'user', content: userQuery };
+    setMessages(prevMessages => [...prevMessages, newUserMessage]);
+
+    const initialAssistantMessage: Message = { role: 'assistant', content: '' }
+    setMessages(prevMessages => [...prevMessages, initialAssistantMessage])
+
+    try {
+      const BackendUrl = process.env.NEXT_PUBLIC_PYTHON_BACKEND_URL;
+      const response = await fetch(`${BackendUrl}/ask`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({ question: userQuery })
+      });
+
+      if (!response.ok || !response.body) {
+        throw new Error(`Failed to fetch response: ${response.status} ${response.statusText}`);
+      }
+
+      const reader = response.body.getReader();
+      const decoder = new TextDecoder();
+
+      let accumulatedContent = ""
+
+      while (true) {
+        const { done, value } = await reader.read();
+        if (done) {
+          console.log('Stream Complete')
+          break;
+        }
+
+        const chunk = decoder.decode(value, { stream: true });
+        const lines = chunk.split('\n').filter(line => line.trim() !== '');
+
+        for (const line of lines) {
+          if (line.startsWith('data: ')) {
+            const jsonString = line.substring(6)
+            try {
+              const parsed = JSON.parse(jsonString)
+
+              if (typeof parsed ==='object' && parsed != null) {
+                if (parsed.type === 'chunk' && typeof parsed.content === 'string') {
+                  accumulatedContent += parsed.content;
+                  setMessages(prevMessages => {
+                      const lastMessage = prevMessages[prevMessages.length - 1];
+                      if (lastMessage && lastMessage.role === 'assistant') {
+                          return [
+                              ...prevMessages.slice(0, -1),
+                              { ...lastMessage, content: accumulatedContent }
+                          ];
```

Comment:
this could be a different function, but let's merge for now. You can clean this up later